events/events.go

```diff
@@ -0,0 +1,137 @@
+package events
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	"github.com/flanksource/commons/logger"
+	"gorm.io/gorm"
+
+	"github.com/flanksource/incident-commander/api"
+	"github.com/flanksource/incident-commander/db"
+	responderPkg "github.com/flanksource/incident-commander/responder"
+)
+
+func ListenForEvents() {
+
+	logger.Infof("Started listening for events")
+
+	// Consume pending events
+	consumeEventsWrapper()
+
+	pgNotify := make(chan bool)
+	go listenToPostgresNotify(pgNotify)
+
+	for {
+		select {
+		case <-pgNotify:
+			consumeEventsWrapper()
+
+		case <-time.After(10 * time.Second):
+			logger.Debugf("timed out waiting for pgNotify")
+		}
+	}
+}
+
+func listenToPostgresNotify(pgNotify chan bool) {
+	ctx := context.Background()
+
+	conn, err := db.Pool.Acquire(ctx)
+	if err != nil {
+		logger.Errorf("Error creating database pool: %v", err)
+	}
+
+	_, err = conn.Exec(ctx, "LISTEN event_queue_updates")
+	if err != nil {
+		logger.Errorf("Error listening to database notify: %v", err)
+	}
+
+	for {
+		_, err = conn.Conn().WaitForNotification(ctx)
+		if err != nil {
+			logger.Errorf("Error waiting for database notifications: %v", err)
+		}
+
+		pgNotify <- true
+	}
+
+}
+
+// TODO: Better function name
+func consumeEventsWrapper() {
```

Comment:
@moshloop @Kaitou786 @ciju 

Need better name suggestions, context is that this function will call `consumeEvents` (which will be renamed to `consumeEvent`) infinitely till there are no more events to consume

Reply:
I would just merge this function into consumeEvents

Reply:
It was previously in the same function but it was a bit complex to read and understand, so decided to split it. I'll use a nested function